druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -774,12 +774,13 @@ protected QuerySpec getQuery(RelDataType rowType, RexNode filter, List<RexNode>
         generator.writeStringField("dataSource", druidTable.dataSource);
         writeField(generator, "granularity", finalGranularity);
         writeField(generator, "dimension", dimensions.get(0));
-        generator.writeStringField("metric", fieldNames.get(collationIndexes.get(0)));
+        generator.writeStringField("metric",
+            collationIndexes == null ? null : fieldNames.get(collationIndexes.get(0)));
         writeFieldIf(generator, "filter", jsonFilter);
         writeField(generator, "aggregations", aggregations);
         writeFieldIf(generator, "postAggregations", postAggs.size() > 0 ? postAggs : null);
         writeField(generator, "intervals", intervals);
-        generator.writeNumberField("threshold", fetch);
+        generator.writeNumberField("threshold", fetch == null ? 0 : fetch);
```

Comment:
too defensive. we would not enter this code if fetch is null. (probably also if collationIndexes is null.) so, better to replace with `assert fetch != null`